game-app/game-core/src/main/java/games/strategy/triplea/delegate/data/CasualtyDetails.java

```diff
@@ -76,13 +78,41 @@ private CasualtyDetails ensureUnitsAreTakenFirst(
             getKilled().stream()
                 .filter(matcher)
                 .collect(Collectors.groupingBy(UnitOwner::new, Collectors.toList())));
-    final List<Unit> damaged =
-        ensureUnitsAreTakenFirst(
-            comparator,
-            targetsGroupedByOwnerAndType,
-            getDamaged().stream()
-                .filter(matcher)
-                .collect(Collectors.groupingBy(UnitOwner::new, Collectors.toList())));
+
+    final List<Unit> damaged = new ArrayList<>();
```

Comment:
Nit, variables should be declared as close to their first usage as possible. Seemingly this variable can be pushed further down and closer to it's usage.

Reply:
The declaration of **damage** can be pushed one statement down.
That increases the distance between declaration and use of **oldTargetUnitsToTakeHits**. Is that really better?

Reply:
Gotchya, I didn't quite see the trade-off. The `damaged` variable did seem to have been declared quite a bit before usage, though I didn't see the nuance behind this.